pkg/controller/secret/secret_controller.go

```diff
@@ -529,11 +555,36 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 			log.Info("INFO: Cluster is not ready; skipping Pager Duty configuration")
 		}
 	}
+
 	if snitchSecretExists {
 		log.Info("INFO: Dead Man's Snitch secret exists")
 		watchdogURL = readSecretKey(r, &request, secretNameDMS, secretKeyDMS)
 	}
 
+	if namespaceCMExists {
+		log.Info("INFO: Managed namespace configMap exists")
+		var managedNamespacesConfig alertmanager.ManagedNamespacesConfig
+		rawManagedNamespaces := readCMKey(r, &request, cmNameManagedNamespaces, cmKeyManagedNamespaces)
+
+		err := yaml.Unmarshal([]byte(rawManagedNamespaces), &managedNamespacesConfig)
+		// yaml.Unmarshal does not throw an error if managedNamespacesConfig does not match -> instead, check that namespace list is empty
+		if (err != nil) || (len(managedNamespacesConfig.Resources.Namespace) == 0) {
```

Comment:
Just a minor observation that we could DRY this code up a bit and have `managedNamespaceList` be set to the default (`redhat-*`, `kube*`, and `openshift-*`) and only check for err == nil here. Appending the managed namespaces would be the exception and you wouldn't need all of these else blocks.